core/src/test/scala/unit/kafka/utils/JaasTestUtils.scala

```diff
@@ -122,6 +136,19 @@ object JaasTestUtils {
   val KafkaScramAdmin = "scram-admin"
   val KafkaScramAdminPassword = "scram-admin-secret"
 
+  val isIBMJdk = System.getProperty("java.vendor").contains("IBM")
+  val serviceName = "kafka"
+
+  def saslConfigs(saslProperties: Option[Properties]): Properties = {
+    val result = saslProperties match {
+      case Some(properties) => properties
+      case None => new Properties
+    }
+    if (isIBMJdk)
+      result.put(KafkaConfig.SaslKerberosServiceNameProp, JaasTestUtils.serviceName)
```

Comment:
The issue is that with IBM's JDK we can't have serviceName in a JaasSection. This field is not standard and J9 throws an exception if it's present. So we have to pass service name as a Kafka config and rely on KerberosLogin.getServiceName() to retrieve the value at runtime.

I agree this should be documented